### DISCUSSION TITLE
How are multicategorical features processed?

### DISCUSSION THREAD

**Discussion by davidfstein** [pyg-team/pytorch-frame#442] [Q&A] [answered]
How are multi-categorical features processed?

**Comment by akihironitta** [comment, marked as answer]
I'd suggest looking at the implementation and its test case as we have no docs around this yet:
https://github.com/pyg-team/pytorch-frame/blob/3710d2f589a6b2891266a81d8f35b7fc4edc1f3f/torch_frame/data/mapper.py#L118-L190
https://github.com/pyg-team/pytorch-frame/blob/3710d2f589a6b2891266a81d8f35b7fc4edc1f3f/test/data/test_mapper.py#L64-L91

**Reply by akihironitta** [reply]
Let me know here if you have any further questions about this!

**Reply by davidfstein** [reply]
So in this example, index of B=0 and index of A=1. So 'A,B' = 1 + 0 = 1 and 'B' = 0. 'C' is not a recognized category so it is also 0? It seems odd to have non-recognized categories be represented with the same value as a recognized category. Am I understanding correctly?

**Reply by akihironitta** [reply]
> 'C' is not a recognized category so it is also 0? 

In short, yes.

Currently, multi-categorical stype columns support `NAStrategy.ZEROS` as in:
https://github.com/pyg-team/pytorch-frame/blob/3710d2f589a6b2891266a81d8f35b7fc4edc1f3f/torch_frame/typing.py#L92-L94

and it means that we always map unknown categories to `0` (e.g., 'C' to `0`):

https://github.com/pyg-team/pytorch-frame/blob/3710d2f589a6b2891266a81d8f35b7fc4edc1f3f/torch_frame/nn/encoder/stype_encoder.py#L120-L121

Mapping unknown categories to `0` means that we map them to the most frequent category as we sort category counts, and we expect `0` is the most frequent category:
https://github.com/pyg-team/pytorch-frame/blob/3710d2f589a6b2891266a81d8f35b7fc4edc1f3f/torch_frame/data/stats.py#L120

We definitely should have a better explaination around this in our docs, but thanks for asking here!

**Reply by davidfstein** [reply]
Ahhh that makes sense! Got it, thank you
